node/src/node/isometrics.cpp

```diff
@@ -0,0 +1,594 @@
+#include <tuple>
+#include <cassert>
+#include <unordered_map>
+
+#include "isometrics.hpp"
+
+#include <boost/range/adaptors.hpp>
+#include <boost/range/algorithm.hpp>
+#include <boost/range/numeric.hpp>
+
+// #define ISOMETRICS_DEBUG
+#undef ISOMETRICS_DEBUG
+
+#ifdef ISOMETRICS_DEBUG
+#define dbg(msg) std::cerr << msg << '\n'
+#define DBG_DUMP_UUIDS(os, logo, container) aux::dump_uuids(os, logo, container)
+#else
+#define dbg(msg)
+#define DBG_DUMP_UUIDS(os, logo, container)
+#endif
+
+namespace cocaine {
+namespace detail {
+namespace service {
+namespace node {
+
+namespace ph = std::placeholders;
+
+namespace conf {
+    constexpr auto PURGATORY_QUEUE_BOUND = 8 * 1024;
+    constexpr auto SHOW_ERRORS_LIMIT = 3;
+
+    // Number of poll iterations without worker metrics updates, used to signal
+    // (post warning to log, etc,) the cases when isolate is active,
+    // but for some reason `forget` to send metrics for active worker without
+    // signaling any error.
+    constexpr auto MISSED_UPDATES_TIMES = 10;
+
+    const std::vector<std::pair<std::string, CounterType>> counter_metrics_names =
+    {
+        // yet abstract cpu load measurement
+        {"cpu", CounterType::instant},
+
+        // memory usage (in bytes)
+        {"vms", CounterType::instant},
+        {"rss", CounterType::instant},
+
+        // running times
+        {"uptime",    CounterType::aggregate},
+        {"user_time", CounterType::aggregate},
+        {"sys_time",  CounterType::aggregate},
+
+        // disk io (in bytes)
+        {"ioread",  CounterType::aggregate},
+        {"iowrite", CounterType::aggregate},
+
+        // TODO: network stats
+
+    };
+}
+
+namespace detail {
+
+    template<typename Val>
+    auto
+    uuid_value(const Val& v) -> std::string;
+
+    template<>
+    auto
+    uuid_value(const std::string& v) -> std::string {
+        return v;
+    }
+
+    template<>
+    auto
+    uuid_value(const dynamic_t& arr) -> std::string {
+        return arr.as_string();
+    }
+
+    template<typename Container>
+    auto
+    dump_uuids(std::ostream& os, const std::string& logo, const Container& arr) -> void {
```

Comment:
Looks like it's some kind of debug helpers. We already have newly introduced display helpers in core, you can use them, besides you can use logger to unify your debug output. If you don't want to use context_t provided logger you can do smth like https://github.com/3Hren/cocaine-plugins/blob/v0.12.13/vicodyn/include/cocaine/vicodyn/debug.hpp
@3Hren Can we think about adding such development debug functionality to core?